R/tar-terra-wrap.R

```diff
@@ -0,0 +1,153 @@
+    filetype <- rlang::arg_match0(filetype, drv$name)
+
+    check_pkg_installed("terra")
+
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    .format_terra_rast_wrap_write <- eval(substitute(function(object, path) {
+        # TODO: provide mapping of major file types to extensions
+        extension <- switch(filetype,
+                            "GTiff" = ".tif",
+                            "GPKG" = ".gpkg",
+                            "")
+        saveRDS(terra::wrapCache(object,
+                                 filename = file.path(cachedir,
+                                                      basename(path),
+                                                      paste0(basename(path),
+                                                             extension)),
+                                 filetype = filetype,
+                                 gdal = gdal,
+                                 overwrite = TRUE),
+                file = path)
+    }, list(cachedir = cachedir, filetype = filetype, gdal = gdal)))
+
+    geotargets_init_cache(name = name)
+
+    # rast_cache_init <- targets::tar_target_raw(
+    #     paste0(name, "_cache_init"),
+    #     str2expression(paste0("normalizePath(file.path(",
+    #                           shQuote(cachedir), ", ",
+    #                           shQuote(name), "))")),
+    #     format = "file"
+    # )
+
+    rast_wrap <- targets::tar_target_raw(
+        name = name,
+        command = command,
+        pattern = pattern,
+        packages = packages,
+        library = library,
+        format = targets::tar_format(
+            read = function(path) terra::unwrap(readRDS(path)),
+            write = .format_terra_rast_wrap_write,
+            marshal = function(object) terra::wrap(object),
+            unmarshal = function(object) terra::unwrap(object)
+        ),
+        repository = repository,
+        iteration = iteration,
+        error = error,
+        memory = memory,
+        garbage_collection = garbage_collection,
+        deployment = deployment,
+        priority = priority,
+        resources = resources,
+        storage = storage,
+        retrieval = retrieval,
+        cue = cue
+    )
+
+    rast_cache_files <- targets::tar_target_raw(
+        paste0(name, "_cache_files"),
+        str2expression(paste0("
+            list.files(
+                file.path(", shQuote(cachedir), ", ", shQuote(name),"),
+                full.names = TRUE,
+                recursive = TRUE
+            )")),
+        format = "file_fast",
+        deps = name
+    )
```

Comment:
I'm questioning whether this target even needs to exist. Unless there is a way to make it be "upstream" of the `wrapCache` target, then it doesn't really serve a purpose.  Invalidating this target will never do anything, and there's no reason to use this target rather than the upstream one in a pipeline.  So maybe this doesn't need to return multiple targets.

Reply:
Agreed, I think it would be ideal to do it in one target. You are right that as implemented it is not doing what I intended.